File: javascript/re-bool.js
const re = /a(?:bc)+?[de]$/;
const str = "regexp123abcbcd";
const repeatCount = 1000000;

const start = Date.now();
for (let i = 0; i < repeatCount; ++i) {
  re.test(str);
}
const end = Date.now();

console.log(end - start);

File: perl/re_bool.pl
use Time::HiRes qw/gettimeofday tv_interval/;

my $re = qr/a(?:bc)+?[de]$/;
my $str = "regexp123abcbcd";
my $repeat_count = 1000000;

my $start = [gettimeofday];
for (my $i = 0; $i < $repeat_count; ++$i) {
  $str =~ $re;
}
my $end = [gettimeofday];
print int(tv_interval($start, $end) * 1000), "\n";
